carbonj.service/src/main/java/com/demandware/carbonj/service/db/points/DataPointRecord.java

```diff
@@ -31,13 +31,14 @@ public static double toValue(byte[] valueBytes)
         return Double.longBitsToDouble( Longs.fromByteArray( valueBytes ) );
     }
 
-    public static int toMetricId(byte[] keyBytes)
+    public static Long toMetricId(byte[] keyBytes)
     {
-        return Ints.fromBytes( keyBytes[0], keyBytes[1], keyBytes[2], keyBytes[3] );
+        return Longs.fromBytes( keyBytes[0], keyBytes[1], keyBytes[2], keyBytes[3], keyBytes[4],
+                keyBytes[5], keyBytes[6], keyBytes[7]);
     }
 
     public static int toTimestamp(byte[] keyBytes)
     {
```

Comment:
Great find! I was looking for this... The values were cut off and looked like not all bytes were considered..